sys/net/gnrc/network_layer/ipv6/nib/_nib-router.c

```diff
@@ -95,8 +95,6 @@ void _snd_rtr_advs(gnrc_netif_t *netif, const ipv6_addr_t *dst, bool final)
                                                  sizeof(addr_str)));
             _snd_ra(netif, dst, final, abr);
         }
-    } else {
-        _snd_ra(netif, dst, final, NULL);
```

Comment:
I think this is the wrong line to remove / to fix. As you pointed out in chat: This can still be reached by a 6LR that does not have multihop prefix and 6LoWPAN context dissemination (`MULTIHOP_P6C`) activated.

Reply:
I am a little bit confused, why GCC then comes to this conclusion though. Maybe it helps to make the `IS_ACTIVE()` check here

https://github.com/RIOT-OS/RIOT/blob/08cdf379083556a19f536fe71ecc6d5399660c4b/sys/net/gnrc/network_layer/ipv6/nib/_nib-router.c#L247-L248

a C-conditional to help GCC figure this out correctly?

Reply:
GCC did figure this out correctly: If `CONFIG_GNRC_IPV6_NIB_MULTIHOP_P6C` is used and a netif is 6ln but not 6lr, the `NULL` pointer dereferencing would very  much happen.

Reply:
> GCC did figure this out correctly: If `CONFIG_GNRC_IPV6_NIB_MULTIHOP_P6C` is used and a netif is 6ln but not 6lr, the `NULL` pointer dereferencing would very much happen.

Ok, but your line removal also removes the `netif` is a 6lr but does not use the `CONFIG_GNRC_IPV6_NIB_MULTIHOP_P6C` case. Then I guess, the check for `CONFIG_GNRC_IPV6_NIB_MULTIHOP_P6C` check either needs a check for the `netif` also being a 6lr (which would be the right check spec-wise) or if `abr` is NULL (which would be the safer check code-wise).

Reply:
So, maybe this is the correct fix? Checking for 6lr in both cases?

GCC's static analysis does indeed notice that the check then is the same in both cases.

Reply:
> check for the `netif` also being a 6lr (which would be the right check spec-wise) or if `abr` is NULL (which would be the safer check code-wise).

OK, I did both now with the check for `abr` being an `assert()`.